### PR TITLE
Feature/alps pub sub match

### PR DESCRIPTION
**Pull request by maciejburda**
Code with tests needed for simple flow:
1. Create main device.
2. Create pub.
3. Create sub.
4. Receive match.

### CODE REVIEW

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 1]
```diff
@@ -30,19 +49,24 @@ open class AlpsManager: MatchMonitorDelegate, ContextManagerDelegate {
...
     // MARK: - Context Manager Delegate
     
     func contextManager(manager monitor: ContextManager, didUpdateLocation: CLLocation) {
-        
+        mobileDevices.findAll { (result) in
+            guard case let .success(mobileDevices) = result else { return }
+            mobileDevices.forEach {
+                guard let deviceId = $0.id else { return }
+                self.locationUpdateManager.tryToSend(location: Location(location: didUpdateLocation), for: deviceId)
+            }
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift:68]: Bit confusing for me. 

- you get all the mobile devices from repo.
- then you send each location of each mobile device.

It means you send location as a sort of batch ? 
Can't those mobile devices update their own location by themselves ?

after watching tryToSend function, this means that all mobiles devices will be at the self.lastLocation of this MAIN device ?

File: AlpsSDK/Repositories/PubSub/PublicationRepository.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//  Created by Maciej Burda on 27/10/2017.
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import Alps
+
+final class PublicationRepository: AsyncCreateable, AsyncReadable, AsyncDeleteable {
+    typealias DataType = Publication
+    private(set) var items = [Publication]()
```
Comment (wenjdu) [inline AlpsSDK/Repositories/PubSub/PublicationRepository.swift:14]: Each device has his own Pub/Sub repository ? 
or main device will keep all the publication from "session"

example : 
Main device creates a Pin device and attach a publication on it. 
Where will this publication be saved ??

**Review by wenjdu — APPROVED** [review]
👍

**Comment by tharpa** [comment]
Guys, be careful with the get all devices calls!  It won't be in the Core API `0.5.0` anymore!  It's a powerful God call and it will be available in the admin API only.  We need to clean up the YAML file @adamsmo .

**Comment by adamsmo** [comment]
ok will remove them from yaml and comment them out in bookkeeper service

File: AlpsSDK/AlpsManager/AlpsManager.swift (hunk 1 above)
Comment (maciejburda) [reply AlpsSDK/AlpsManager/AlpsManager.swift:68]: Real life scenario is that you have only one device (your mobile phone).

But Rafał mentioned that we shouldn't disallow creating multiple mobile devices for now.

File: AlpsSDK/Repositories/PubSub/PublicationRepository.swift (hunk 2 above)
Comment (maciejburda) [reply AlpsSDK/Repositories/PubSub/PublicationRepository.swift:14]: Take a look at ` func createPublication(publication: Publication, for deviceWithId: String? = nil, completion: ((Publication?) -> Void)? = nil)`

It's convenice method in `AlpsManager`.

All publications are stored in publication repostiory. If you want to pull those releated to device you need to filter by `deviceId`.

Comment (wenjdu) [reply AlpsSDK/Repositories/PubSub/PublicationRepository.swift:14]: got it, thanks

**Pull request merged**